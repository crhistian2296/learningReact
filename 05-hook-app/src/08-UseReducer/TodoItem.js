import React from 'react';

const TodoItem = ({ todo, index, handleDelete, handleToggle }) => {
  return (
    <div
      key={todo.id}
      className='list-group-item d-flex align-content-center justify-content-between'
    >
      <button
        className={`btn toggle text ${todo.done === true && 'complete'}`}
        onClick={() => handleToggle(todo.id)}
      >
        {index + 1}. {todo.description}
      </button>
      <button className='btn btn-danger' onClick={() => handleDelete(todo.id)}>
        Eliminar
      </button>
    </div>
  );
};

export default TodoItem;
