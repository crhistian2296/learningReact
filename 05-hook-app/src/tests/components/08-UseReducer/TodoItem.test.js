import { shallow } from 'enzyme';
import TodoItem from '../../../08-UseReducer/TodoItem';
import { demoTodos } from '../../fixtures/demoTodos';

describe('Pruebas en <todoListitem/>', () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  const wrapper = shallow(
    <TodoItem
      todo={demoTodos[0]}
      index={0}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  test('Debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de llamar a la funcion handleDelete', () => {
    wrapper.find('.btn-danger').simulate('click');
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });

  test('Debe de llamar a la funcion handleToggle', () => {
    wrapper.find('.toggle').simulate('click');
    expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
  });

  test('Debe de mostrar el texto del item correctamente', () => {
    const p = wrapper.find('.text').text();
    // `{0 + 1}. {todo.description}`
    expect(p).toBe(`1. ${demoTodos[0].description}`);
  });

  test('Debe de tener la clase "complete" si todo.done es true', () => {
    const todo = demoTodos[0];
    todo.done = true;

    const wrapper = shallow(<TodoItem todo={todo} />);
    expect(wrapper.find('.toggle').hasClass('complete')).toBeTruthy();
  });
});
