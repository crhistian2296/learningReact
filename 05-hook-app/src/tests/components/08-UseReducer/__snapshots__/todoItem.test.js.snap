// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <todoListitem/> Debe de mostrarse correctamente 1`] = `
<div
  className="list-group-item d-flex align-content-center justify-content-between"
  key="1"
>
  <button
    className="btn toggle text false"
    onClick={[Function]}
  >
    1
    . 
    Aprender Typescript
  </button>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Eliminar
  </button>
</div>
`;
